### PR TITLE
Add a metric to track number of online lookups

### PR DESCRIPTION
**Pull request by ssrihari**
**Story card:** [ch4211](https://app.clubhouse.io/simpledotorg/story/4211/create-a-dashboard-to-monitor-metrics-relating-to-performance-of-the-online-lookup-feature)

- Increment a metric for every patient looked up
- Add the retention type in the metric name
- Add state and user-id in tags as metadata

### CODE REVIEW

File: app/controllers/api/v4/patients_controller.rb [hunk 1]
```diff
@@ -15,7 +15,11 @@ def lookup
     trigger_audit_log(patients)
     render(
       json: Oj.dump({
-        patients: patients.map { |patient| Api::V4::PatientLookupTransformer.to_response(patient, retention(patient)) }
+        patients: patients.map do |patient|
+          retention = retention(patient)
+          Api::V4::PatientLookupTransformer.to_response(patient, retention)
```
Comment (harimohanraj89) [inline app/controllers/api/v4/patients_controller.rb:20]: We have to return this from the block.

Comment (ssrihari) [reply app/controllers/api/v4/patients_controller.rb]: Fixed in bcf610e 🤦

Comment (harimohanraj89) [reply app/controllers/api/v4/patients_controller.rb]: Classic 😅

File: app/controllers/api/v4/patients_controller.rb [hunk 2]
```diff
@@ -15,7 +15,11 @@ def lookup
     trigger_audit_log(patients)
     render(
       json: Oj.dump({
-        patients: patients.map { |patient| Api::V4::PatientLookupTransformer.to_response(patient, retention(patient)) }
+        patients: patients.map do |patient|
+          retention = retention(patient)
+          Statsd.instance.increment("OnlineLookup.#{retention[:type]}", tags: [current_state.name, current_user.id])
```
Comment (harimohanraj89) [inline app/controllers/api/v4/patients_controller.rb:20]: I suspect `current_state` and `current_user` are guaranteed to be present for an authenticated request? If so, :100:

**Review by harimohanraj89 — APPROVED** [review]
Sweet sweet metrics.

**Pull request merged**